consensus/safety-rules/src/error.rs

```diff
@@ -32,6 +33,9 @@ pub enum Error {
         last_voted_round: Round,
         proposal_round: Round,
     },
+
+    #[error("Serialization error: {:?}", error)]
+    SerializationError { error: String },
```

Comment:
This error type is serializable. Lcs is not and it isn't expected that most other error types would be.

Implementing a from doesn't seem to work either:

impl From<lcs::Error> for Error {
    fn from(error: lcs::Error) -> Self {
        Self::SerializationError(format!("{}", error))
    }
}

error[E0308]: mismatched types
  --> consensus/safety-rules/src/remote.rs:50:17
   |
34 |     pub fn handle_message(&mut self, input_message: Vec<u8>) -> Result<Vec<u8>, Error> {
   |                                                                 ---------------------- expected `std::result::Result<std::vec::Vec<u8>, error::Error>` because of return type
...
50 |                 lcs::to_bytes(&self.internal.sign_timeout(&timeout))
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `error::Error`, found enum `libra_canonical_serialization::error::Error`
   |
   = note: expected type `std::result::Result<_, error::Error>`
              found type `std::result::Result<_, libra_canonical_serialization::error::Error>`

Please elaborate.

Reply:
Seems we can ? which would force an implicit conversion and then wrap with ok

So...

Ok(value?)
seems to work fine... good call.